components/AddMemberModal.tsx

```diff
@@ -1,36 +1,122 @@
+import { useAnchorWallet, useConnection } from '@solana/wallet-adapter-react'
 import { FormikErrors, useFormik } from 'formik'
+import { useAppSelector } from '../hooks/useAppSelector'
+import { selectCluster } from '../redux/features/wallet/walletSlice'
+import { useState } from 'react'
 import { useRef } from 'react'
+import { FanoutClient } from '@glasseaters/hydra-sdk'
+import { PublicKey, Transaction } from '@solana/web3.js'
+import FormStateAlert, { FormState } from './FormStateAlert'
+
+type AddMemberModalProps = {
+  hydraWallet: any
+}
 
 interface FormValues {
-  pubKey: string
+  pubkey: string
   shares: number
 }
 
-const AddMemberModal = () => {
+const AddMemberModal = ({ hydraWallet }: AddMemberModalProps) => {
   let toggleRef = useRef<HTMLInputElement>(null)
 
   const initialValues = {
-    pubKey: '',
-    shares: 0
+    pubkey: '',
+    shares: '',
   }
 
-  const onSubmit = (values: any) => {
+  const { connection } = useConnection()
+  const cluster = useAppSelector(selectCluster)
+  const wallet = useAnchorWallet()
+
+  const [formState, setFormState] = useState('idle' as FormState)
+  const [errorMsg, setErrorMsg] = useState('')
+  const [logs, setLogs] = useState([])
+
+  const onSubmit = async (values: any, { resetForm }) => {
     console.log('submitted', values)
     // add the wallet member here
+    if (!wallet) {
+      setFormState('error')
+      setErrorMsg('Please connect your wallet!')
+      return
+    }
+
+    try {
+      setLogs([])
+      const fanoutSdk = new FanoutClient(connection, wallet)
+
+      // Prepare transaction
+      const tx = new Transaction()
+      const ixAddMember = await fanoutSdk.addMemberWalletInstructions({
+        fanout: new PublicKey(hydraWallet.pubkey),
+        membershipKey: new PublicKey(values.pubkey),
+        shares: values.shares,
+      })
+      tx.add(...ixAddMember.instructions)
+
+      // Sign transaction using user's wallet
+      tx.recentBlockhash = (await connection.getLatestBlockhash()).blockhash
+      tx.feePayer = wallet.publicKey
+      const txSigned = await wallet.signTransaction(tx)
+
+      //Send API request
+      const res = await fetch('/api/addUser', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          tx: txSigned?.serialize().toString('base64'),
+          memberPubkey: values.pubkey,
+          shareCount: values.shares,
+          walletPubKey: hydraWallet.pubkey,
+          cluster,
+        }),
+      })
 
-    toggleRef.current!.checked = false
+      if (res.status === 200) {
+        setFormState('success')
+
+        //wait 2 seconds before closing the form and reload the page
+        setTimeout(function () {
+          toggleRef.current!.checked = false
+          location.reload()
+        }, 3200)
```

Comment:
I think we do not need to refresh the page or close the modal as that would be bad user experience especially if the user wants to add multiple members. If you want to refresh the members table you can revalidate the wallet details [SWR](https://swr.vercel.app/) cache to force a reload:

```typescript
// Add this at the top of the AddMemberModal component
const { mutate } = useSWRConfig()
.
.
.
// Add this on success instead of the page reloading logic
mutate(`/api/wallets/${hydraWallet.pubkey}?cluster=${cluster}`)
```